src/paper/tasks.py

```diff
@@ -870,97 +870,26 @@ def pull_arxiv_papers_directly():
-#                 raw_authors = result.get("authorships", [])
-#                 concepts = result.get("concepts", [])
-#                 abstract = rebuild_sentence_from_inverted_index(
-#                     result.get("abstract_inverted_index", {})
-#                 )
-
-#                 doi_paper_check = Paper.objects.filter(doi_svf=SearchQuery(pure_doi))
-#                 url_paper_check = Paper.objects.filter(
-#                     Q(url_svf=SearchQuery(oa_pdf_url))
-#                     | Q(pdf_url_svf=SearchQuery(oa_pdf_url))
-#                 )
-#                 if doi_paper_check.exists() or url_paper_check.exists():
-#                     # This skips over the current iteration
-#                     continue
-
-#                 data = {
-#                     "doi": pure_doi,
-#                     "url": url,
-#                     "raw_authors": format_raw_authors(raw_authors),
-#                     "title": title,
-#                     "paper_title": title,
-#                     "paper_publish_date": result.get("publication_date", None),
-#                     "is_open_access": oa.get("is_oa", None),
-#                     "oa_status": oa.get("oa_status", None),
-#                     "pdf_license": source.get("license", None),
-#                     "external_source": source.get("display_name", None),
-#                     "abstract": abstract,
-#                 }
-#                 if oa_pdf_url and check_url_contains_pdf(oa_pdf_url):
-#                     data["pdf_url"] = oa_pdf_url.strip()
-
-#                 try:
-#                     paper = Paper(**data)
-#                     paper.full_clean()
-#                     paper.save()
-
-#                     concept_names = [
-#                         concept.get("display_name", "other")
-#                         for concept in concepts
-#                         if concept.get("level", 0) == 0
-#                     ]
-#                     potential_hubs = []
-#                     for concept_name in concept_names:
-#                         potential_hub = Hub.objects.filter(name__icontains=concept_name)
-#                         if potential_hub.exists():
-#                             potential_hub = potential_hub.first()
-#                             potential_hubs.append(potential_hub)
-#                             hub_ids.add(potential_hub.id)
-#                     paper.hubs.add(*potential_hubs)
-#                     download_pdf.apply_async((paper.id,), priority=4, countdown=4)
-#                 except Exception as e:
-#                     sentry.log_error(e)
-#         arxiv_works = open_alex.get_data_from_source(arxiv_id, yesterday, page=i + 1)
-#     reset_unified_document_cache(
-#         hub_ids=hub_ids,
-#         document_type=["paper"],
-#         filters=[NEW],
-#     )
-#     return total_works
+@periodic_task(
+    run_every=crontab(minute=0, hour="2"), priority=3, queue=QUEUE_PULL_PAPERS
```

Comment:
Maybe this should be daily instead of every 2 hours? It puts a lot of strain on our worker server

Reply:
This is Cronjob for every day at 2am right?

Reply:
Oh yeah, I misread it.